app/scripts/modules/core/src/managed/ArtifactDetail.tsx

```diff
@@ -160,25 +179,43 @@ export const ArtifactDetail = ({
                     noticeType="error"
                   />
                 )}
-                {resourcesByEnvironment[environmentName]
-                  .filter(resource => shouldDisplayResource(name, type, resource))
-                  .map(resource => (
-                    <div key={resource.id} className="flex-container-h middle">
-                      {state === 'deploying' && (
-                        <div
-                          className={classNames(
-                            'resource-badge flex-container-h center middle sp-margin-s-right',
-                            state,
-                          )}
+                {statefulConstraints && (
+                  <ConstraintCards
+                    constraints={statefulConstraints}
+                    application={application}
+                    environment={environmentName}
+                    version={version}
+                  />
+                )}
+                {statelessConstraints && (
+                  <ConstraintCards
+                    constraints={statelessConstraints}
+                    application={application}
+                    environment={environmentName}
+                    version={version}
+                  />
+                )}
+                <div className="sp-margin-l-top">
```

Comment:
nothing changed in this block, just whitespace from wrapping it for proper margins